### ISSUE TITLE
load old messages upon request

### ISSUE THREAD

**Issue by rabble** [planetary-social/planetary-ios#909]
Users in the app need to be able to choose whether or not messages older than 6 months are loaded from go-ssb in go sqlite. This right now is set to 6 months and users can't choose. This should be a toggle in advanced settings. 

This is checked at ViewDatabase:fillMessages 

            /* This is the don't put older than 6 months in the db. */
            if isOldMessage(message: msg) &&
                (msg.content.type != .contact &&
                msg.content.type != .about &&
                msg.author != currentUser) {
                skipped += 1
                print("Skipped(\(msg.content.type) \(msg.key)%)")
                continue
            }
            
  It needs to check if the user WANTS old messages.

**Comment by mplorentz** [comment]
Yes! I really want this feature. I picture the setting as a slider that allows users to choose how long they want to keep posts: i.e. 6 months, 1 year, 2 years, forever.

We will also need an accompanying system to honor this setting. I think ideally we would do this in the background without interrupting the user's normal use of the app. We will need to delete old posts from SQLite or copy old posts from GoBot to SQLite depending on how the setting is changed.

**Comment by rabble** [comment]
Probably something we fix in Scuttlego.

**Comment by boreq** [comment]
This is now unblocked but not planned for 2.0.0 as far as I understand.

**Comment by boreq** [comment]
This is now unblocked but not planned for 2.0.0 as far as I understand.
